prow/pubsub/subscriber/subscriber.go

```diff
@@ -58,6 +61,78 @@ type prowCfgClient interface {
 	GetPostsubmitsStatic(identifier string) []config.Postsubmit
 }
 
+type InRepoConfigCacheGetter struct {
+	CookieFilePath string
+	CacheSize      int
+	Agent          *config.Agent
+	mu             sync.Mutex
+	GithubOptions  flagutil.GitHubOptions
+	DryRun         bool
+
+	CacheMap map[string]*config.InRepoConfigCache
+}
+
+func (irc *InRepoConfigCacheGetter) getCache(cloneURI, host string) (*config.InRepoConfigCache, error) {
+	// No repo is cloned in getCache, Since this function should happen fast it is safe to lock the whole function.
+	irc.mu.Lock()
+	defer irc.mu.Unlock()
+	if irc.CacheMap == nil {
+		irc.CacheMap = map[string]*config.InRepoConfigCache{}
+	}
+
+	var key string
+	// We are using github with IRC
+	if irc.GithubOptions.Host != "" && (irc.GithubOptions.TokenPath != "" || irc.GithubOptions.AppPrivateKeyPath != "") {
+		key = irc.GithubOptions.Host
+		// We are using Gerrit with IRC
```

Comment:
this comment is misplaced, should be in the closure below, same with the next comment. Not a blocker, feel free address in followup PR